### DISCUSSION TITLE
GLSPProjectionView not working as expected

### DISCUSSION THREAD

**Discussion by msaquib17** [eclipse-glsp/glsp#801] [Q&A]
`@inject(FrontendApplication) protected readonly application: FrontendApplication;

constructor(@inject(TYPES.IActionDispatcher) protected actionDispatcher: GLSPActionDispatcher) {
    super();
    this.inputElement = document.createElement("input");
}

protected initializeContents(containerElement: HTMLElement): void {
    // do nothing
}

async show(root: Readonly<SModelRoot>, ...contextElementIds: string[]) {
    super.show(root, ...contextElementIds);
    if (this.isEditorEnabled()) {
        if (this.label) {
             console.log(this)
             if (this.application.shell.activeWidget instanceof DiagramWidget) {
                const diagramWidget = this.application.shell.activeWidget;
                const sourceUri=diagramWidget.uri.toString();`

In my code the diagramWidget variable is turning out to be null when I use GLSPProjectionView in `configureModelElement(context, "graph", GLSPGraph, GLSPProjectionView)`. However, it works fine when I use SGraphView as in `configureModelElement(context, "graph", GLSPGraph, GLSPProjectionView)`. Is there something wron with GLSPProjectionView or I am missing something?

**Comment by tortmayr** [comment]
HI @msaquib17,
sorry for the late reply. At first glance your code looks corret to me so this might be a bug.
I will try to reproduce the issue and then come back to you.

**Reply by msaquib17** [reply]
Hi @tortmayr , is there any workaround that exists for this situation? We want to use GLSPProjectionView and at the same time need the file uri in the variable sourceUri in the code.

**Reply by tortmayr** [reply]
@msaquib17 
I tried to reproduce your issue based on the snipped you provided and everything works fine for me. The diagram widget is always defined independently of which view (projection vs SGraph) I used. I'm also pretty sure that there is some other underlying issue. The type of graph view should have no effect on whether the diagram widget is defined or not.

Based on the code snipped I assume that you are working with an UI extension. Could you please debug and check the values of `shell.activateWidget` and `shell.currentWidget`. Maybe another widget is still in Focus? This could happen if the extension is activated to early.
